src/engine/game/party.cpp

```diff
@@ -183,9 +183,22 @@ void Party::setPartyLeader(int npc) {
     }
     if (memberIdx == 0) return;
 
+    setPartyLeaderByIndex(memberIdx);
+}
+
+void Party::setPartyLeaderByIndex(int index) {
+    if (index == 0) {
```

Comment:
`index` could be negative. Also, logging here is redundant IMO, unless this method will be called from a script:

```cpp
if (index < 1 || index >= members.size()) return;
```